### PR TITLE
Yarn v3

### PR DESCRIPTION
**Pull request by adonesky1**
Upgrades package manager from Yarn v1 to Yarn v3.

Followed the [migration guide](https://yarnpkg.com/getting-started/migration) + some of the example set in https://github.com/MetaMask/core/pull/846 for @metamask/controller

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,3 +1,8 @@
 * text=auto
 
 yarn.lock linguist-generated=false
+
+# yarn v3
+# See: https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored
+/.yarn/releases/** binary
+/.yarn/plugins/** binary
```
Comment (Gudahtt) [inline .gitattributes:8]: Nit: We can restore the trailing newline here

```suggestion
/.yarn/plugins/** binary

```

File: .gitignore [hunk 2]
```diff
@@ -1,58 +1,17 @@
...
+
+# yarn v3 (w/o zero-install)
+# See: https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored
+.pnp.*
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
```
Comment (Gudahtt) [inline .gitignore:17]: Nit: Another missing trailing newline
```suggestion
!.yarn/versions

```

File: .gitignore [hunk 3]
```diff
@@ -1,58 +1,17 @@
-# Logs
-logs
-*.log
-npm-debug.log*
-yarn-debug.log*
-yarn-error.log*
```
[the remark this answers is not in the document]
Comment (adonesky1) [reply .gitignore:6]: done [here](https://github.com/MetaMask/eth-trezor-keyring/pull/150/commits/0ecbe4f22f4ad9865411f9695fdc60986e6b7de0)

**Comment by Gudahtt** [comment]
There are two lines needing an update in the `lint-test.yml` workflow, as done here: https://github.com/MetaMask/core/pull/846/files#diff-b168d1081eb1f2b453a20c874165b094602f6984903a444b2762650a72e80001

**Comment by adonesky1** [comment]
> There are two lines needing an update in the lint-test.yml workflow, as done here

done [here](https://github.com/MetaMask/eth-trezor-keyring/pull/150/commits/16a8f01005f21b36e85660ee6c5806961d1e09ee) 

@Gudahtt

File: .github/workflows/lint-test.yml [hunk 4]
```diff
@@ -29,7 +29,7 @@ jobs:
         with:
           path: ${{ steps.yarn-cache-dir.outputs.YARN_CACHE_DIR }}
           key: yarn-cache-${{ runner.os }}-${{ steps.yarn-version.outputs.YARN_VERSION }}-${{ hashFiles('yarn.lock') }}
-      - run: yarn --frozen-lockfile
+      - run: yarn --immutable
       - run: yarn allow-scripts
```
Comment (Gudahtt) [inline .github/workflows/lint-test.yml:33]: Sorry, 3 lines I guess, I missed this one
```suggestion
```

Comment (adonesky1) [reply .github/workflows/lint-test.yml:33]: Whoops I should've caught that

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**